applications: Refactored Matter bridge storage implementation (#14988)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

   The :kconfig:option:`CONFIG_BRIDGE_BT_SCAN_TIMEOUT_MS` kconfig option to set the scan timeout.

* Updated the implementation of the persistent storage to leverage ``NonSecure``-prefixed methods from the common Persistent Storage module.
* Changed data structure of information stored in the persistent storage to use less settings keys.

>   The new structure uses approximately 2.5 times less memory, and provides a new field to store user-specific data.

Author: And how about "uses approximately 40% of the memory used by the old structure"?

>   The new structure uses approximately 2.5 times less memory, and provides a new field to store user-specific data.

Reviewer: That should also work.

>   The new structure uses approximately approximately 40% of the memory used by the old structure, and provides a new field to store user-specific data.

Reviewer: ```suggestion
  The new structure uses approximately 40% of the memory used by the old structure, and provides a new field to store user-specific data.
```

   The :kconfig:option:`CONFIG_BRIDGE_BT_SCAN_TIMEOUT_MS` kconfig option to set the scan timeout.

* Updated the implementation of the persistent storage to leverage ``NonSecure``-prefixed methods from the common Persistent Storage module.
* Changed data structure of information stored in the persistent storage to use less settings keys.

>   It allowed to use approximately 2.5 times less memory and provided new field to store user specific data.
>   Backward compatibility is kept by a internal dedicated method that automatically detects older data format and performs data migration to the new representation.

Reviewer: ```suggestion
  The new structure uses approximately 2.5 times less memory, and provides new field to store user-specific data.
  
  Backward compatibility is kept by using an internal dedicated method that automatically detects the older data format and performs data migration to the new representation.
```
